### DISCUSSION TITLE
Linguist doesn't recognize code in github-pages

### DISCUSSION THREAD

**Discussion by JohnyP36** [github-linguist/linguist#6281] [Q&A] [answered]
I am running my own website via a github repo *(repo name = [johnyp36.github.io](https://github.com/JohnyP36/JohnyP36.github.io))* and I added the `.gitattributes` file. But still it does not show which languages are used. 

<details><summary>I put this in the `.githattributes` file</summary>

```gitattributes
*.css linguist-language=css linguist-detectable
*.html linguist-language=hmtl linguist-detectable
*.xhtml linguist-language=hmtl linguist-detectable
*.js linguist-language=js linguist-detectable
docs/ linguist-detectable
docs/ -linguist-documentation
docs/ -linguist-generated

```

</details>

**Comment by lildude** [comment, marked as answer]
You only need one line and it needs to look like this

```gitattributes
docs/** -linguist-documentation
```

Why? Because `docs/` is not recursive so the override would only apply to that directory. Using `docs/*` would apply the override only to the files within `docs/` but not any of its subdirectories, hence we need `docs/**` to recurse all the way down.

This is not specific to Linguist. It's how the path matching works with `.gitattributes` as detailed in the [`.gitattributes` docs](https://www.git-scm.com/docs/gitattributes):

> The rules by which the pattern matches paths are the same as in `.gitignore` files (see [gitignore[5]](https://www.git-scm.com/docs/gitignore)), with a few exceptions:
> 
> - negative patterns are forbidden
> - patterns that match a directory do not recursively match paths inside that directory (so using the trailing-slash `path/` syntax is pointless in an attributes file; use `path/**` instead)

**Reply by JohnyP36** [reply]
I now works.  The `**` after 'docs/' solved the problem

This  what I have now. 

```gitattributes
docs/** -linguist-documentation
*.xhtml linguist-language=hmtl linguist-detectable
*.woff2 linguist-language=css linguist-detectable
```

**Comment by JohnyP36** [comment]
I also added

```gitattributes
.github/** -linguist-documentation
*.yml linguist-language=YAML linguist-detectable
```

But it seems linguist does not document what's in the `.github` folder. Although in the [documentation](https://github.com/github/linguist/blob/master/lib/linguist/documentation.yml) does not mention it.

**Reply by lildude** [reply]
That's because `.github` is considered vendored [here](https://github.com/github/linguist/blob/c1c34e5260797b4d598f5ec76f19723bfc5a1894/lib/linguist/vendor.yml#L386-L387) so if you really want this analysed and showing in your statistics you'll need to change your `.github` line to:

```gitattributes
.github/** -linguist-vendored
```
